### PR TITLE
Enable rlc sample generation option

### PR DESCRIPTION
**Pull request by MaryGao**
Enable rlc sample generation option including below files:

- package.json
- tsconfig.json

Also add the minimal sample code to create the client in file `${clientFileName}Sample`

### CODE REVIEW

File: src/restLevelClient/generateRestLevel.ts [hunk 1]
```diff
@@ -76,9 +77,11 @@ export async function generateRestLevelClient() {
   generateIndexFile(project);
 
   generateTopLevelIndexFile(model, project);
-  if (generateTest) {
+  if (generateSample) {
     generateSampleEnv(project);
```
Comment (MaryGao) [inline src/restLevelClient/generateRestLevel.ts:81]: @qiaozha I change the condition from `generateTest` to `generateSample`, could you help confirm any reason here to use `generateTest` to generate file `sample.env`?

Comment (qiaozha) [reply src/restLevelClient/generateRestLevel.ts:81]: yeah, change to generateSample should be correct

File: tsconfig.browser-test.json [hunk 2]
```diff
@@ -21,7 +21,15 @@
     "newLine": "LF",
     "outDir": "./test-browser",
     "rootDir": "./test",
-    "useUnknownInCatchVariables": false
+    "useUnknownInCatchVariables": false,
+    "paths": {
+      "@msinternal/body-string-rest": [
```
Comment (MaryGao) [inline tsconfig.browser-test.json:26]: Note here I will ping the package path to ensure files in `samples-dev` could build successfully

File: src/generators/static/tsConfigFileGenerator.ts [hunk 3]
```diff
@@ -34,7 +34,7 @@ const restLevelTsConfigInAzureSdkForJs = {
   "include": ["src/**/*.ts"]
 };
 
-const restLevelTsConfigNotInAzureSdkForJs ={
+const restLevelTsConfigNotInAzureSdkForJs: Record<string, any> = {
```
Comment (joheredi) [inline src/generators/static/tsConfigFileGenerator.ts:37]: I wonder if there is a type for this that we can import from the typescript package

File: src/generators/static/tsConfigFileGenerator.ts [hunk 4]
```diff
@@ -81,6 +81,9 @@ export function generateTsConfig(project: Project) {
     highLevelTsConfig.include.push("samples-dev/**/*.ts");
     highLevelTsConfig.compilerOptions["paths"] = {};
     highLevelTsConfig.compilerOptions["paths"][clientPackageName] = ["./src/index"];
+    restLevelTsConfig.include.push("samples-dev/**/*.ts");
```
Comment (joheredi) [inline src/generators/static/tsConfigFileGenerator.ts:84]: do we need to check for generateSamples?

File: src/restLevelClient/samples/generateSamples.ts [hunk 5]
```diff
@@ -0,0 +1,23 @@
+export interface SampleGroup {
```
Comment (joheredi) [inline src/restLevelClient/samples/generateSamples.ts:1]: Don't forget to add the license header in all new files

// Copyright (c) Microsoft Corporation.
// Licensed under the MIT License.

File: test/rlcIntegration/generated/bodyStringRest/package.json [hunk 6]
```diff
@@ -107,5 +107,11 @@
   },
   "browser": {
     "./dist-esm/test/public/utils/env.js": "./dist-esm/test/public/utils/env.browser.js"
+  },
+  "//sampleConfiguration": {
+    "productName": "",
```
Comment (joheredi) [inline test/rlcIntegration/generated/bodyStringRest/package.json:112]: Why is this being generated as empty string?

**Review by joheredi — APPROVED** [review]
Looks good, I left a few comments

File: src/generators/static/tsConfigFileGenerator.ts (hunk 3 above)
Comment (MaryGao) [reply src/generators/static/tsConfigFileGenerator.ts:37]: I didn't find this `type` after searching it. Let me know if you know similar types.

File: src/generators/static/tsConfigFileGenerator.ts (hunk 4 above)
Comment (MaryGao) [reply src/generators/static/tsConfigFileGenerator.ts:84]: It's already under the condition of generateSample

File: test/rlcIntegration/generated/bodyStringRest/package.json (hunk 6 above)
Comment (MaryGao) [reply test/rlcIntegration/generated/bodyStringRest/package.json:112]: Due to the description in package detail is empty and will improve this in next pr

**Pull request merged**